### ISSUE TITLE
Pluralize no longer works with the new register-helper method for Handlebars

### ISSUE THREAD

**Issue by jrhe** [stefanpenner/ember-inflector#75]
Helpers registered using `Ember.Handlebars.helper` receive a number of positional args instead of an array of arguments as we are now expecting. This results in pluralize helper receiving count as its params argument rather than an array of `[count, word]`.

https://github.com/stefanpenner/ember-inflector/blob/5e80bac26f6bc7ff7fc50386f3bee08468c601e8/addon/lib/helpers.js#L43

I have attempted to pass a wrapped version of `helperFunction` to `Ember.Handlebars.helper` but haven't been able to get this working as ember-cli seems to be pushing two lots of ember-inflector into my compiled files, one of which I can't seem to edit.

![screenshot 2015-03-26 18 44 59](https://cloud.githubusercontent.com/assets/4038905/6854287/43d893bc-d3e8-11e4-921b-eec8a8f01d77.png)

**Comment by stefanpenner** [comment]
which version of ember are you using?

**Comment by jrhe** [comment]
1.9.1

**Comment by ryanjm** [comment]
I'm not including `ember-inflector` directly, but from what I can tell in the source code, it is being included by something else (I think?).

I'm getting the following deprecation warning:

```
DEPRECATION: Using Ember.HTMLBars._registerHelper is deprecated. Helpers (even dashless ones) are automatically resolved. [deprecation id: ember-htmlbars.register-helper]
```

And I believe it is pointing to this [line](https://github.com/stefanpenner/ember-inflector/blob/1440332cfa967e19ddb2e4a3a0531b4c7f22cbda/addon/lib/helpers/pluralize.js#L21).

Am I looking in the right place? Looking at that code it looks like your [makeHelper](https://github.com/stefanpenner/ember-inflector/blob/1440332cfa967e19ddb2e4a3a0531b4c7f22cbda/addon/lib/utils/make-helper.js#L3) function shouldn't cause any issues.

I'm trying to figure out where this deprecation warning is coming from. Thanks!

| Lib | Version |
|----------|--------------|
| Ember Inspector | 1.8.3 |
| Ember | 1.13.6 |
| Ember Data | 1.13.7 |
| jQuery | 1.11.3 |

<img width="434" alt="screen_shot_2015-08-07_at_5_53_22_am" src="https://cloud.githubusercontent.com/assets/90015/9135267/e018b470-3cc8-11e5-9bee-13020e57a333.png">

**Comment by jcope2013** [comment]
@ryanjm update ember data to 1.13.8, that is the problem, it needed a inflector dependency bump as it uses inflector

**Comment by ryanjm** [comment]
Thanks @jcope2013. I just updated to `1.13.8` and still have the same issue. I'll dig into it more this weekend, but thanks for the direction.

**Comment by ryanjm** [comment]
Just to document my progress incase anyone else runs into this. I've not been able to fix it yet. I've cleared cache / made sure everything is up to date. But it looks like [EmberTwiddle](http://ember-twiddle.com/) is having the same issue, even after updating it to `1.13.8` also. I think something else is missing.

**Comment by adam-knights** [comment]
@stefanpenner I think maybe this should be re-opened (or a new issue made) - if I goto EmberTwiddle like @ryanjm and edit twiddle.json from `1.13.7` to be `1.13.8` for Ember data I see the exact same deprecation warnings on _registerHelper and makeBoundHelper as I get in my Ember CLI 1.13.7 with Ember 1.13.6 and Data 1.13.8

**Comment by engwan** [comment]
I'm currently upgrading my app and trying to remove deprecations and I'm stuck at this too.

I'm also at Ember 1.13.6 and Ember data 1.13.8 like @adam-knights

**Comment by jcope2013** [comment]
changing it to http://builds.emberjs.com/tags/v1.13.8/ember-data.js makes it go away? can you try that?

**Comment by adam-knights** [comment]
@jcope2013 Agreed - that url in twiddle makes the deprecations go away. So I downloaded both that file and the cdnjs one and compared to the one I've got locally from bower - my local file is an exact match of the cdnjs one, ie the one that causes the warnings on twiddle.

**Comment by adam-knights** [comment]
So I guess the question is why is 1.13.8 of ember-data.js from either bower or cdnjs giving these warnings while the build tag file from emberjs.com file isn't?

**Comment by jcope2013** [comment]
@adam-knights can you file an issue on ember data? i have a feeling they published 1.13.7 as 1.13.8 or the wrong 1.13.8 on bower

**Comment by adam-knights** [comment]
Submited issue as request https://github.com/emberjs/data/issues/3635
